mPower2/mPower2/View Controllers/TrackingViewController.swift

```diff
@@ -88,84 +109,173 @@ class TrackingViewController: UIViewController {
+        // We have to adjust the constraints on the headerContentView to either center vertically in its
+        // superview, or pin the top and bottom to its superview. This will determine how big the image is,
+        // among other things.
+        
+        func adjustHeaderView(to height: CGFloat) {
+            if let headerView = tableView.tableHeaderView {
+
+                var frame = headerView.frame
+                if height != frame.size.height {
+                    frame.size.height = height
+                    headerView.frame = frame
+                    tableView.tableHeaderView = headerView
+                }
+            }
+        }
+        
+        if completedTaskGroups.count > 0 {
+            let height = shouldShowActionBar ? kHeaderViewHeightSmall : kHeaderViewHeightMedium
+            adjustHeaderView(to: height)
+            if let heightConstraint = headerContentView.rsd_constraint(for: .height, relation: .equal) {
+                NSLayoutConstraint.deactivate([heightConstraint])
+            }
+            
+            // Inset the top of the tableView
+            let inset = shouldShowActionBar ? kTableViewVerticalPadding : kTableViewTopInsetNoActionBar
+            tableView.contentInset = UIEdgeInsets(top: inset, left: 0.0, bottom: 0.0, right: 0.0)
+        }
+        else {
+            adjustHeaderView(to: tableView.bounds.height)
+            let contentHeight = [tableView.bounds.height - (2 * kTableViewVerticalPadding), kHeaderViewHeightLarge].min()
+            headerContentView.rsd_makeHeight(.equal, contentHeight ?? 0)
+            
+            // Inset the top of the tableView jto 0
+            tableView.contentInset = UIEdgeInsets(top: 0.0, left: 0.0, bottom: 0.0, right: 0.0)
+        }
+
+        // Adjust our contentOffset to account for contentInset
+        tableView.setContentOffset(CGPoint(x: 0, y: -1 * tableView.contentInset.top), animated: false)
+        
+        // Adjust headerView bottom constraint (the padding between bottom label and bottom of view)
+        headerViewBottomConstraint.constant = kTableViewVerticalPadding
+    }
+    
+    // MARK: Model
+    
+    func taskGroups() -> [RSDTaskGroup] {
+        // TODO: jbruhin 5-1-18 obtain task model from the proper source?? Model also needs more tasks
+        // and tasks may need more data, like icon images
+        let taskGroups: [RSDTaskGroup] = {
+            let activeTaskGroup : RSDTaskGroup = {
+                let taskInfos = MCTTaskIdentifier.all().map { MCTTaskInfo($0) }
+                var taskGroup = RSDTaskGroupObject(with: "Measuring", tasks: taskInfos)
+                taskGroup.title = "Measuring"
+                return taskGroup
+            }()
+            let trackingTaskGroup : RSDTaskGroup = {
+                
+                var taskInfos = [RSDTaskInfoObject]()
+                ["Symptoms", "Medication", "Triggers", "Medication", "Medication"].forEach { (identifier) in
```

Comment:
Assuming this is a test of cell layout? (having Medication listed multiple time)